python/ray/data/read_api.py

```diff
@@ -384,6 +387,7 @@ def read_datasource(
     # Compute the number of blocks the read will return. If the number of blocks is
     # expected to be less than the requested parallelism, boost the number of blocks
     # by adding an additional split into `k` pieces to each read task.
+    additional_split_factor = None
```

Comment:
One small issue. For the new code path, `get_read_tasks` has actually created the read tasks. But the read tasks are only used for the following calculations and then discarded. 
I'm wondering if we can move the following calculation code to the reader. 
So here we only create the reader, but not the read tasks. Also we don't need to expose this `additional_split_factor` to the operator.

Reply:
yes, that's the major weird thing here. The other code path `LazyBlockList` has all code paths depending on `List[ReadTask]`, so I don't spend time to refactoring `LazyBlockList`. Shall we just delete `DatasetPipeline` and `LazyBlockList`/`BlockList` during 2.8?

Reply:
that's fine. can you leave a todo here?

Reply:
SG, added.